### ISSUE TITLE
condition for children does not work

### ISSUE THREAD

**Issue by greyblake** [joshbuddy/jsonpath#38]
It was fixed for array here: https://github.com/joshbuddy/jsonpath/pull/24

But still does not work, if a child is not an array:

``` ruby
json = <<-JSON
{
  "store": {
    "book": {
      "category": "reference",
      "author": "Nigel Rees",
      "title": "Sayings of the Century",
      "price": 8.95
    }
  }
}
JSON

pp JsonPath.new("$..book[?(@.price<10)]").on(json) # => []
# But expected to return the book

# But checking price directly works also
pp JsonPath.new("$..book.price[?(@<10)]").on(json) # => [8.95]
```

**Comment by greyblake** [comment]
P.S. I you decide to merge the PR, can you please publish a new version on rubygems?:)

**Comment by Skarlso** [comment]
So... There hasn't been any action on this. I'm guessing the author is gone, or something. Can somebody take over the curation of this gem? 

I'd be happy too.

**Comment by greyblake** [comment]
@Skarlso I confirm it'x fixed in 0.7.0. Thanks a lot!

**Comment by Skarlso** [comment]
Awesome buddy! Thanks very much for the confirm. 👍
